include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -528,34 +528,24 @@ def TT_TransOp : TT_Op<"trans", [Pure,
 //
 // SPMD Ops
 //
-def TT_GetProgramIdOp : TT_Op<"get_program_id", [Pure]> {
-    let arguments = (ins TT_ProgramDim:$axis);
-
-    let results = (outs I32:$result);
-
-    let assemblyFormat = "$axis attr-dict `:` type($result)";
 
-    let extraClassDeclaration = [{
-      int32_t getAxisAsInt() {
-        return static_cast<int32_t>(getAxis());
-      }
-    }];
-}
-
-def TT_GetNumProgramsOp : TT_Op<"get_num_programs", [Pure]> {
+class program_id_inst <string name> : TT_Op<name, [Pure]> {
     let arguments = (ins TT_ProgramDim:$axis);
 
     let results = (outs I32:$result);
 
     let assemblyFormat = "$axis attr-dict `:` type($result)";
 
     let extraClassDeclaration = [{
-      int32_t getAxisAsInt() {
-        return static_cast<int32_t>(getAxis());
-      }
+        int32_t getAxisAsInt() {
+            return static_cast<int32_t>(getAxis());
+        }
     }];
 }
 
+def TT_GetProgramIdOp   : program_id_inst<"get_program_id">;
+def TT_GetNumProgramsOp : program_id_inst<"get_num_programs">;
+
```

Comment:
To be honest I find the previous format more readable even if a bit more verbose

Reply:
I think I get your point. I changed this to avoid separating two `def`s, which might lead one unchanged while updating another, e.g. fixed in #3346.

I can undo this if you think it's better.

Reply:
yes please